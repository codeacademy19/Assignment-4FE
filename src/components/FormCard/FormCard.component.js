import React, { Component } from 'react';
import {
  SafeAreaView, ImageBackground, Text, View, Image, TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './FormCard.css';


class FormCard extends Component {
  render() {
    const { name, time } = this.props;

    return (
      <View style={styles.container}>
        <View style={styles.formDetail}>
          <Text style={styles.formName}>{name}</Text>

        </View>
        <View style={styles.horizantalLine} />
        <View style={styles.formTimeView}>
          <Text style={styles.formTime}>
Created On:
            {' '}
            {time}
          </Text>
        </View>
      </View>
    );
  }
}

// FormCard.defaultProps = {
//   name: 'default',
//   time: '15/01/01',
// };

FormCard.propTypes = {
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
};

export default FormCard;
