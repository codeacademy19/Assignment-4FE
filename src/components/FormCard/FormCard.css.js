import { StyleSheet } from 'react-native';
import { FORM_TIME, FORM_NAME } from '../../constants';

export default StyleSheet.create({

  container: {
    margin: 10,
    marginTop: 20,
    padding: 10,
    flex: 1,
    width: '80%',
    height: 250,
    backgroundColor: 'white',
  },
  formName: {
    marginTop: 10,
    marginLeft: 10,
    fontSize: FORM_NAME,
    height: '70%',
  },
  horizantalLine: {
    marginTop: 10,
    borderStyle: 'solid',
    borderColor: 'black',
    borderWidth: 0.5,
  },
  formTime: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    marginTop: 10,
    paddingBottom: 10,
    fontSize: FORM_TIME,
  },
  formDetail: {
    textAlign: 'left',
  },
  formTimeView: {
    textAlign: 'right',
    alignItems: 'flex-start',
  },
});
