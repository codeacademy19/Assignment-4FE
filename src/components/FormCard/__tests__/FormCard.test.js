import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FormCard from '../FormCard.component';


describe('CardView component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <FormCard name="default" time="15/0/1" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
