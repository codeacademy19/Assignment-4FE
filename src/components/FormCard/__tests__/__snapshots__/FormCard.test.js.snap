// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard component renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 250,
      "margin": 10,
      "marginTop": 20,
      "padding": 10,
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 21,
          "height": "70%",
          "marginLeft": 10,
          "marginTop": 10,
        }
      }
    >
      default
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 0.5,
        "marginTop": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "textAlign": "right",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 16,
          "marginTop": 10,
          "paddingBottom": 10,
        }
      }
    >
      Created On:
       
      15/0/1
    </Text>
  </View>
</View>
`;
